CONTRIBUTING.md

```diff
@@ -0,0 +1,96 @@
+# HERMES development workflow
+
+## Preamble
+
+> **Branching is cheap!**
+>
+> Work in small packets, put up small pull requests!
+>
+> Aim for quick turnaround times!
+
+## Branching
+
+We loosely follow a mixture of [GitHubFlow](https://docs.github.com/en/get-started/quickstart/github-flow) and [GitFlow](https://nvie.com/posts/a-successful-git-branching-model/):
+
+`main`
+    - Stable branch
+    - Merges come only from `develop` or a hotfix branch (i.e., when something needs to be fixed in "production")
+
+`develop`
+    - Unstable development branch
+    - Last "stable" is tagged: tag before attempting to break something
+
+`feature/<describe-feature>` (including issue id when exists: `feature/62-fix-broken-thing`)
+    - Branch from last tag on `develop`
+    - Feature branches may pick up "older" branches if they are not to complicated
+    - Combined branches must all start from the same tag
+
+### Pull requests
+
+- As soon as you have made 1 commit in a feature branch, put up a *draft* pull request
+- Keep pull requests small :skull: 
+- :warning: No pre-emptive reviews on PR drafts, **unless** the PR author @-mentions with this specific request
+- When you think you're done, mark PR ready for review, see below (Merge Process)
+
+## Merge Process (into `develop`)
+
+- Create PR from `feature/...` against `develop` (PR author)
+- Describe work in initial comment (PR author)
+    - Issue reference (`#n`)
+    - What does the new code do
+    - Optional: What should reviewers look at specifically
+    - Information on how to review:
+        - E.g.
+          ```bash
+          pip install ./
+          pytest test/
+          ```
+- Request review (PR author)
+    - Eligible reviewers:
+        - Python code: @led02, @sdruskat, @jkelling, #TODO
+        - Documentation:
+            - Workflow: @all
+            - Project: @all
+- Review (at least 1 reviewer)
+    - Follow instructions in PR
+    - Review thoroughly beyond instructions
+    - Comments / change suggestions inline in files
```

Comment:
**thought**: should we recommend "conventional comments"?

Reply:
Yes... but, no.

While I like the basic idea of [conventional commits](https://www.conventionalcommits.org). However, since we used it in one of my projects, I realized: You cannot "just use" conventional commits.
Even tough, CC provides a format / template on how commit messages should look like, there is very little semantics given there. Especially, the `type`s are not pre-defined.

I would encourage everyone to use the format, but making it mandatory would require more work.

Reply:
This will only ever work when we define our own types, and everyone always adheres to the semantics & format. Which is unrealistic in my experience.

I think @poikilotherm's comment relates to the PR comments that the actual docs sections is about. In which case I'd say: no... but yes. I think it's nice form to briefly comment what kind of change you're suggesting, to make it easier for the PR author to assess. I.e., *Fix wording.* or *Explain thing in more detail.* is not cluttering things, while also making sure that there is an explanation.

Reply:
Huh, didn't know that one. Good suggestions, but I'd stick with keeping it optional. 

"Non-blocking" is expressed in how you finish your review (Approve, Comment, Request changes). 

The labels can be useful though, I'll give using them a shot in the future.

Reply:
> You cannot "just use" conventional commits. [...], there is very little semantics given there. Especially, the `type`s are not pre-defined.

FWIW: I'm using this [cheat sheet](https://github.com/iamturns/til/blob/master/dev/git/conventional-commits-cheatsheet.md) all the time and it serves me quite well.

Reply:
**praise:** I did not know that but like ita lot. Maybe a good recommandation.
**troll:** However, it will make it much more visible when I'm about to troll around again...